src/main/java/org/opensearch/sdk/ssl/DefaultSslKeyStore.java

```diff
@@ -0,0 +1,834 @@
+        return sslContext;
+    }
+
+    private void logExplanation(Exception e) {
+        if (ExceptionUtils.findMsg(e, "not contain valid private key") != null) {
+            log.error(
+                "Your keystore or PEM does not contain a key. "
+                    + "If you specified a key password, try removing it. "
+                    + "If you did not specify a key password, perhaps you need to if the key is in fact password-protected. "
+                    + "Maybe you just confused keys and certificates."
+            );
+        }
+
+        if (ExceptionUtils.findMsg(e, "not contain valid certificates") != null) {
+            log.error("Your keystore or PEM does not contain a certificate. Maybe you confused keys and certificates.");
+        }
+    }
+
+    private static void checkPath(String keystoreFilePath, String fileNameLogOnly) {
+
+        if (keystoreFilePath == null || keystoreFilePath.length() == 0) {
+            throw new OpenSearchException("Empty file path for " + fileNameLogOnly);
+        }
+
+        if (Files.isDirectory(Paths.get(keystoreFilePath), LinkOption.NOFOLLOW_LINKS)) {
+            throw new OpenSearchException("Is a directory: " + keystoreFilePath + " Expected a file for " + fileNameLogOnly);
+        }
+
+        if (!Files.isReadable(Paths.get(keystoreFilePath))) {
+            throw new OpenSearchException(
+                "Unable to read "
+                    + keystoreFilePath
+                    + " ("
+                    + Paths.get(keystoreFilePath)
+                    + "). Please make sure this files exists and is readable regarding to permissions. Property: "
+                    + fileNameLogOnly
+            );
+        }
+    }
+
+    @Override
+    public String getSubjectAlternativeNames(X509Certificate cert) {
+        String san = "";
+        try {
+            Collection<List<?>> altNames = cert != null && cert.getSubjectAlternativeNames() != null
+                ? cert.getSubjectAlternativeNames()
+                : null;
+            if (altNames != null) {
+                Collection<List<?>> sans = new ArrayList<>();
+                for (List<?> altName : altNames) {
+                    Integer type = (Integer) altName.get(0);
+                    sans.add(altName);
+                    // otherName requires parsing to string
+                    // if (type == 0) {
+                    // List<?> otherName = getOtherName(altName);
+                    // if (otherName != null) {
+                    // sans.add(Arrays.asList(type, otherName));
+                    // }
+                    // } else {
+                    // sans.add(altName);
+                    // }
+                }
+                san = sans.toString();
+            }
+        } catch (CertificateParsingException e) {
+            log.error("Issue parsing SubjectAlternativeName:", e);
+        }
+
+        return san;
+    }
+
+    // private List<String> getOtherName(List<?> altName) {
```

Comment:
Why is this commented out?

This originally came to the security plugin to fix a bug with certificate hot reloads: https://github.com/opensearch-project/security/pull/1369

I chose to comment this out because it would have required a dependency on bouncycastle.